static/js/markdown.js

```diff
@@ -102,8 +102,16 @@ export function apply_markdown(message) {
     const options = {
         userMentionHandler(mention, silently) {
             if (mention === "all" || mention === "everyone" || mention === "stream") {
-                message.mentioned = true;
-                return `<span class="user-mention" data-user-id="*">@${_.escape(mention)}</span>`;
+                if (!silently) {
+                    message.mentioned = true;
+                    return `<span class="user-mention" data-user-id="*">@${_.escape(
+                        mention,
+                    )}</span>`;
+                }
+
+                return `<span class="user-mention silent" data-user-id="*">${_.escape(
+                    mention,
+                )}</span>`;
```

Comment:
I think we want to rather than having the whole syntax be an if/else just have a variable `classes = "user-mention"` that has ` silent` added to it for the silent case.  That way, we avoid duplicating the escaping logic (etc.), which would increase the risk of future bugs.

Reply:
Done.